tsdb/engine/tsm1/wal.go

```diff
@@ -366,6 +366,13 @@ func (w *WriteWALEntry) Encode(dst []byte) ([]byte, error) {
 
 	for k, v := range w.Values {
 
+		// Make sure we have enough space in our buf before copying.  If not,
+		// grow the buf.
+		if len(dst[:n])+2+len(k)+len(v)*8+4 > len(dst) {
```

Comment:
So this change is because the buffer was insufficiently big for even the first batch, right?